integrations/amazon_sagemaker/src/haystack_integrations/components/generators/amazon_sagemaker/sagemaker.py

```diff
@@ -138,27 +162,47 @@ def from_dict(cls, data) -> "SagemakerGenerator":
         """
         Deserialize the dictionary into an instance of SagemakerGenerator.
         """
+        deserialize_secrets_inplace(
+            data["init_parameters"],
+            ["aws_access_key_id", "aws_secret_access_key", "aws_session_token", "aws_region_name", "aws_profile_name"],
+        )
         return default_from_dict(cls, data)
 
-    def warm_up(self):
+    @classmethod
+    def get_aws_session(
+        cls,
+        aws_access_key_id: Optional[str] = None,
+        aws_secret_access_key: Optional[str] = None,
+        aws_session_token: Optional[str] = None,
+        aws_region_name: Optional[str] = None,
+        aws_profile_name: Optional[str] = None,
+        **kwargs,
+    ):
         """
-        Initializes the SageMaker Inference client.
+        Creates an AWS Session with the given parameters.
+        Checks if the provided AWS credentials are valid and can be used to connect to AWS.
+
+        :param aws_access_key_id: AWS access key ID.
+        :param aws_secret_access_key: AWS secret access key.
+        :param aws_session_token: AWS session token.
+        :param aws_region_name: AWS region name.
+        :param aws_profile_name: AWS profile name.
+        :param kwargs: The kwargs passed down to the service client. Supported kwargs depend on the model chosen.
+            See https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters.html.
+        :raises AWSConfigurationError: If the provided AWS credentials are invalid.
+        :return: The created AWS session.
         """
-        boto3_import.check()
         try:
-            session = boto3.Session(
-                aws_access_key_id=os.getenv(self.aws_access_key_id_var),
-                aws_secret_access_key=os.getenv(self.aws_secret_access_key_var),
-                aws_session_token=os.getenv(self.aws_session_token_var),
-                region_name=os.getenv(self.aws_region_name_var),
-                profile_name=os.getenv(self.aws_profile_name_var),
-            )
-            self.client = session.client("sagemaker-runtime")
-        except Exception as e:
-            msg = (
-                f"Could not connect to SageMaker Inference Endpoint '{self.model}'."
-                f"Make sure the Endpoint exists and AWS environment is configured."
+            return boto3.Session(
+                aws_access_key_id=aws_access_key_id,
+                aws_secret_access_key=aws_secret_access_key,
+                aws_session_token=aws_session_token,
+                region_name=aws_region_name,
+                profile_name=aws_profile_name,
             )
+        except BotoCoreError as e:
+            provided_aws_config = {k: v for k, v in kwargs.items() if k in AWS_CONFIGURATION_KEYS}
```

Comment:
`kwargs` will never contain any of the fields in `AWS_CONFIGURATION_KEYS` as they're all explicit in `get_aws_session`.

This should be changed.